### PR TITLE
py-atpublic: new recipe

### PR DESCRIPTION
**Pull request by Andrew-Dunning-NNL**
add py-atpublic recipie

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-atpublic/package.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+    homepage = "https://public.readthedocs.io"
+    pypi     = "atpublic/atpublic-2.1.2.tar.gz"
+
+    version('2.1.2', sha256='82a2f2c0343ac67913f67cdee8fa4da294a4d6b863111527a459c8e4d1a646c8')
+    version('2.1.1', sha256='fa1d48bcb85bbed90f6ffee6936578f65ff0e93aa607397bd88eaeb408bd96d8')
+
+    depends_on('python@3.6:', type=('build', 'run'))
+    depends_on('py-setuptools', type='build')
+    depends_on('py-typing-extensions@:3.8', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-atpublic/package.py]:
```suggestion
    depends_on('py-typing-extensions', when='^python@:3.7.999', type=('build', 'run'))
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-atpublic/package.py (hunk 1 above)
Comment (Andrew-Dunning-NNL) [reply var/spack/repos/builtin/packages/py-atpublic/package.py]: Corrected py-typing-extensions dependency

File: var/spack/repos/builtin/packages/py-atpublic/package.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+    homepage = "https://public.readthedocs.io"
+    pypi     = "atpublic/atpublic-2.1.2.tar.gz"
+
+    version('2.1.2', sha256='82a2f2c0343ac67913f67cdee8fa4da294a4d6b863111527a459c8e4d1a646c8')
+    version('2.1.1', sha256='fa1d48bcb85bbed90f6ffee6936578f65ff0e93aa607397bd88eaeb408bd96d8')
+
+    depends_on('python@3.6:', type=('build', 'run'))
+    depends_on('py-setuptools', type='build')
+    depends_on('py-typing-extensions', when='^python@:3.8', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-atpublic/package.py]:
```suggestion
    depends_on('py-typing-extensions', when='^python@:3.7.999', type=('build', 'run'))
```
It isn't needed in 3.8, version ranges are inclusive in Spack

**Review by adamjstewart — APPROVED** [review]

**Comment by Andrew-Dunning-NNL** [comment]
@adamjstewart Hello Adam! Did this package need something else to be merged?

**Pull request merged**